### PR TITLE
[hotfix][docs] Fixing multiple 404 links in docs

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* Addressing multiple 404s

## Brief change log

* Updated references to documentation in various places

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: docs/content/docs/dev/table/sql/queries/match_recognize.md [hunk 1]
```diff
@@ -267,7 +267,7 @@ look at the [event stream navigation](#pattern-navigation) section.
 ### Aggregations
 
 Aggregations can be used in `DEFINE` and `MEASURES` clauses. Both
-[built-in]({{{{< ref "docs/dev/table/functions/systemFunctions" >}}) and custom
+[built-in]({{{{< ref "docs/dev/table/functions/systemfunctions" >}}) and custom
```
Comment (gaoyunhaii) [inline docs/content/docs/dev/table/sql/queries/match_recognize.md:270]: Here seems to have too much `{`?

File: docs/content/docs/dev/table/sql/queries/match_recognize.md [hunk 2]
```diff
@@ -267,7 +267,7 @@ look at the [event stream navigation](#pattern-navigation) section.
...
 [user defined]({{{{< ref "docs/dev/table/functions/udfs" >}}) functions are supported.
```
Comment (gaoyunhaii) [inline docs/content/docs/dev/table/sql/queries/match_recognize.md:271]: And also here, have too much `{`.

**Review by gaoyunhaii — COMMENTED** [review]
Very thanks @MartijnVisser for opening the PR! I have one small comment.

File: docs/content/docs/dev/table/sql/queries/match_recognize.md (hunk 1 above)
Comment (MartijnVisser) [reply docs/content/docs/dev/table/sql/queries/match_recognize.md]: Thanks for that, fixed it!

File: docs/content/docs/dev/table/sql/queries/match_recognize.md (hunk 2 above)
Comment (MartijnVisser) [reply docs/content/docs/dev/table/sql/queries/match_recognize.md]: Thanks for that, fixed it too!

**Review by gaoyunhaii — APPROVED** [review]
Very thanks @MartijnVisser for the PR! LGTM~

**Pull request closed without merging**